llvm/lib/Target/RISCV/RISCVInstrInfoXCheri.td

```diff
@@ -1162,34 +1164,40 @@ def : InstAlias<"c.cincoffsetimm4cspn $rd, $rs1, $imm",
                 (C_CIncOffsetImm4CSPN GPCRC:$rd, CSP:$rs1, uimm10_lsb00nonzero:$imm), 0>;
 def : InstAlias<"c.cincoffsetimm16csp $rd, $imm",
                 (C_CIncOffsetImm16CSP CSP:$rd, simm10_lsb0000nonzero:$imm), 0>;
-def : MnemonicAlias<"c.addi4spn", "c.cincoffsetimm4cspn">;
-def : MnemonicAlias<"c.addi16sp", "c.cincoffsetimm16csp">;
-def : MnemonicAlias<"c.lw", "c.clw">;
-def : MnemonicAlias<"c.sw", "c.csw">;
-def : MnemonicAlias<"c.sc", "c.csc">;
-def : MnemonicAlias<"c.lc", "c.clc">;
-def : MnemonicAlias<"c.lwsp", "c.clwcsp">;
-def : MnemonicAlias<"c.swsp", "c.cswcsp">;
-def : MnemonicAlias<"c.scsp", "c.csccsp">;
-def : MnemonicAlias<"c.lcsp", "c.clccsp">;
-def : MnemonicAlias<"c.jr", "c.cjr">;
-def : MnemonicAlias<"c.jalr", "c.cjalr">;
+def : InstAlias<"c.addi4spn $rd, $rs1, $imm", (C_CIncOffsetImm4CSPN GPCRC:$rd, CSP:$rs1, uimm10_lsb00nonzero:$imm)>;
+def : InstAlias<"c.addi16sp $rd, $imm", (C_CIncOffsetImm16CSP CSP:$rd, simm10_lsb0000nonzero:$imm), 0>;
+def : InstAlias<"c.clw $rd, ${imm}(${rs1})", (C_CLW GPRC:$rd, GPCRC:$rs1, uimm7_lsb00:$imm), 0>;
+def : InstAlias<"c.csw $rs2, ${imm}(${rs1})", (C_CSW GPRC:$rs2, GPCRC:$rs1, uimm7_lsb00:$imm), 0>;
+def : InstAlias<"c.clwcsp $rd, ${imm}(${rs1})", (C_CLWCSP GPRNoX0:$rd, CSP:$rs1, uimm8_lsb00:$imm), 0>;
+def : InstAlias<"c.cswcsp $rs2, ${imm}(${rs1})", (C_CSWCSP GPR:$rs2, CSP:$rs1, uimm8_lsb00:$imm), 0>;
+def : InstAlias<"c.cjalr $rs1", (C_CJALR GPCRNoC0:$rs1), 0>;
+def : InstAlias<"c.cjr $rs1", (C_CJR GPCRNoC0:$rs1), 0>;
 } // Predicates = [HasCheri, HasCheriRVC, HasStdExtC, IsCapMode]
 let Predicates = [HasCheri, HasCheriRVC, HasStdExtC, IsRV64, IsCapMode] in {
-def : MnemonicAlias<"c.ld", "c.cld">;
-def : MnemonicAlias<"c.sd", "c.csd">;
-def : MnemonicAlias<"c.ldsp", "c.cldcsp">;
-def : MnemonicAlias<"c.sdsp", "c.csdcsp">;
+def : InstAlias<"c.cld $rd, ${imm}(${rs1})", (C_CLD GPRC:$rd, GPCRC:$rs1, uimm8_lsb000:$imm), 0>;
```

Comment:
Changed these to InstAlias instead of MnemonicAlias since otherwise the error messages we get for bad register arguments are "missing capmode" instead of "illegal operand"